File: selenium-comparison/spec/tests/auth02_invalid_password_spec.rb
require_relative '../spec_helper'

# Mirrors: tests/auth/auth-02-invalid-password.spec.ts
# No retries — invalid login may trigger APEX throttling on repeated runs.

RSpec.describe '3 — Authentication' do
  subject(:login) { LoginPage.new(@driver) }

  it 'AUTH-02 — Login with invalid password shows error message' do
    login.navigate_to_login

    login.fill_credentials('invalid@example.com', 'WrongPassword123!')
    login.submit

    expect(@driver.current_url).to match(/login_desktop/)
    expect(login.error_visible?).to be(true)

    msg = login.error_message
    expect(msg).to be_truthy
    expect(msg.length).to be > 0
  end
end

File: tests/auth/auth-02-invalid-password.spec.ts
// spec: specs/sanzu.plan.md
// seed: fixtures/fixtures.ts

import { test, expect } from '../../fixtures/fixtures';

test.describe.configure({ retries: 0 });

test.describe('3 — Authentication', () => {
  test('AUTH-02 — Login with invalid password shows error message', async ({ loginPage, page }) => {
    // Step 1: Navigate to /login_desktop via loginPage.goto().
    await loginPage.goto();

    // Step 2: Call loginPage.loginForm.fill() with invalid credentials.
    await loginPage.loginForm.fill('invalid@example.com', 'WrongPassword123!');

    // Step 3: Call loginPage.loginForm.submit().
    await loginPage.loginForm.submit();

    // Step 4: Assert page stays on /login_desktop (no redirect to /home).
    await expect(page).toHaveURL(/login_desktop/);

    // Step 4: Assert error message is visible.
    await expect(loginPage.loginForm.errorMessage).toBeVisible();

    // Step 4: Assert getError() returns a truthy string.
    const error = await loginPage.loginForm.getError();
    expect(error).toBeTruthy();
  });
});
